File: Lab 4/InvertTreeMain.py
class TreeNode(object):

     def __init__(self, val=0, left=None, right=None):

         self.val = val

         self.left = left

         self.right = right

class InvertTreeMain:
    def invertTree(self, root):
        if root is None:
            return None
        
        left = self.invertTree(root.left)
        right = self.invertTree(root.right)
        
        root.left = right
        root.right = left
        
        return root

if __name__ == "__main__":
    tree = InvertTreeMain()
    tree.root = TreeNode(12)
    tree.root.left = TreeNode(8)
    tree.root.right = TreeNode(15)
    tree.invertTree(tree.root)

    print(tree.root.left.val)
    print(tree.root.right.val)
    print(tree.root.val)

File: Lab 4/invertTreeMain .java

public class invertTreeMain {

	TreeNode root;

	public TreeNode invertTree(TreeNode root) {

		if (root.left != null && root.right != null) {
			TreeNode left = invertTree(root.left);
			TreeNode right = invertTree(root.right);
			root.left = right;
			root.right = left;
		
		}

		return root;

	}

	public static void main(String[] args) {

		invertTreeMain tree = new invertTreeMain();
		tree.root = new TreeNode(12);
		tree.root.left = new TreeNode(8);
		tree.root.right = new TreeNode(15);
		tree.invertTree(tree.root);

		System.out.println(tree.root.left.val);
		System.out.println(tree.root.right.val);
		System.out.println(tree.root.val);
	}

}
